soc/silabs/Kconfig

```diff
@@ -318,4 +320,6 @@ config SOC_GECKO_CUSTOM_RADIO_PHY
 	  management, sending and receiving packets on radio phy. User has
 	  to provide the radio_config.c and radio_config.h files for the phy.
 
-endif # SOC_FAMILY_SILABS_S0 || SOC_FAMILY_SILABS_S1 || SOC_FAMILY_SILABS_S2
+endif # !SOC_FAMILY_SILABS_SIM3
+
+endif # SOC_FAMILY_SILABS_S{0,1,2} || SOC_FAMILY_SILABS_SIM3
```

Comment:
I am not super fond of all this boolean arithmetic. I believe it would be easier if we applied the condition in the inner Kconfig:
```
diff --git i/soc/silabs/Kconfig w/soc/silabs/Kconfig
index dd098de9559..c20fef18c9b 100644
--- i/soc/silabs/Kconfig
+++ w/soc/silabs/Kconfig
@@ -2,10 +2,10 @@
 # Copyright (c) 2018 Gil Benkoe
 # SPDX-License-Identifier: Apache-2.0

-if SOC_FAMILY_SILABS_S0 || SOC_FAMILY_SILABS_S1 || SOC_FAMILY_SILABS_S2
-
 rsource "*/Kconfig"

+if SOC_FAMILY_SILABS_S0 || SOC_FAMILY_SILABS_S1 || SOC_FAMILY_SILABS_S2
+
 config SOC_GECKO_SDID
        int
        help
diff --git i/soc/silabs/silabs_s0/Kconfig w/soc/silabs/silabs_s0/Kconfig
index dfc253eb424..6baa33bc1ac 100644
--- i/soc/silabs/silabs_s0/Kconfig
+++ w/soc/silabs/silabs_s0/Kconfig
@@ -2,12 +2,12 @@
 #
 # SPDX-License-Identifier: Apache-2.0

+if SOC_FAMILY_SILABS_S0
+
 config SOC_FAMILY_SILABS_S0
        select HAS_SEGGER_RTT if ZEPHYR_SEGGER_MODULE
        select BUILD_OUTPUT_HEX

-if SOC_FAMILY_SILABS_S0
-
 rsource "*/Kconfig"

 endif # SOC_FAMILY_SILABS_S0
...
```

Reply:
Adapted accordingly (I think).